notebooks/2018-11-27-Value-when-other-is-max.ipynb

```diff
@@ -414,125 +408,104 @@
6frEwoId2771/5Px75fuyXnfo9890S9v8bZXvm7s4Iz7nn3IWK+cz58Vf5+UeZIfSWrXdnmD+842Ec/xR73U4L6ryZ93emSic27XUo8jrtYdurqdDvpF0Y/72v0XVuT3qz482goAAAAAiIULSQAAAAC1x5XgTxZm1t7MRpvZDDN718z2NLN+ZvaamU01s/+YWdu09r8xs5lm9p6ZHZK2fXCwbaaZXZK2fWszG29mH5jZfWbWNNjeLCjPDOq7ZxsrF5IAAAAAUB6ukTTGOddLUj9J70q6VdIlzrkdJT0s6SJJMrPekk6Q1EfSYEk3mlljM2ss6QZJh0rqLenEoK0k/U3S1c65npK+kjQ82D5c0lfOuW0lXR20i8RkOzkoZCayXIUzkWHfdsk9M7li65V5GZMkrXaNvXK+M5Ppi/LWQl6yksTJRBZay2s3ZJgKmpf8VSjPemz+ug5nIMOZyXAmMpv0TGQxdXq5Vaz2d324jVfu2+nzfA4nZ5WSiczmzBE/88qNo390SIMKl5GsFMM6jPPKU57YrkQjya9C5iCTWPJ1S6+cJDNJJrL6WJlNFRPcadxH0jBJcs6tkrTKzLaXtO4f4DOSxkr6g6QjJd3rnFspabaZzZS07j+zM51zs4J+75V0pJm9K2l/ST8K2oyQdKmkm4K+Lg22j5Z0vZmZi5hQhzuSAAAAAFB6PSQtkHSHmU0ys1vNrJWkdyQdEbQZKqlr8LqLpE/T9p8TbMu0fRNJi51za0Lbvb6C+iVB+4y4kAQAAABQW5wk54r/R+poZhPS/pyRNqqNJO0i6Sbn3M6Slkm6RNKpks4xs4mS2khaNwVzfY9NugZsj+orIx5tBQAAAIDiWBix/MccSXOcc+OD8milspF/kHSwJJnZdpIOS2vfNW3/LSXNDV7Xt32hpPZmtlFw1zG9/bq+5pjZRpLaSVoU9YVwIZmDbtdPybntJ+fuFFm/onPuD2O3mlM52cxwZjLdld97wCtPXLZ13o4bzkwOaDU71v5Tlue+zlulKuS6kdlcMSfz+n9J7dr+Y6+c77Uic5Wel6xPITOUcdaNDMu2jmQ24Qzlkx/5Y7ni84PXv35+cnS+dcAT0Z+bSXxvp/e9cpJM5KeLNo63w+BvvGKrMW0afGx8V6WsG1lOwutGHvLHX61/Pfayq7y68LqRp3d72SsvXNNW+ZJtLch8CmcmI/Uk0xs26tm969n6SNHHUa2cc5+b2admtr1z7j1JB0iabmadnXPzzayRpN9LujnY5TFJd5vZVZK2kNRT0htK3V3saWZbS/pMqQl5fuScc2b2vKTjJN0r6WRJj6b1dbKk14L656LykRIXkgAAAABqULlNthM4T9KoYFmOWZJOkXSSma37DfBDku6QJOfcNDO7X9J0SWskneOcWytJZnauUpPyNJZ0u3NuWrD/xZLuNbM/S5ok6bZg+22S7gom7Fmk1MVnJC4kAQAAAKAMOOcmSwo/+npN8Ke+9n+R9Jd6tj8p6cl6ts/Shpld07evUGoin5xxIQkAAACg9pTnHcmKwYVknoXzlO9ftmOD+1q2ZfS/7kJmKK1pftdnLJZw/jJuZjJd+pqSUnmvKxkea7q9Wr9XxJEUT6kykXGFM5RJMpMrN1vtlcM5x/DakIW0387TvXJ6JrKcvDrFX6Ov3eZfR7Yv5LqS1bJ2ZDrLtm5kyNJxnTLWtR60IOFoEDZqbHSONJyLjJItExln3chiZiLz6e5pu0XW/6jPm0UaCVAeuJAEAAAAUFNMZZuRrBisIwkAAAAAiIULSQAAAABALDzaWsHCGcq6DqtKNJJoF73qTwB1RQHXlQzL1ncTq8wsaBwTlvfwyru2nFWwY31T17xgfVeLRX9enb1RBp212CvPn9feK4czk7bGz1G7jXJ/hmfa4dd75f+dnzmHK0kLVrXOWHf8QD9ffP8b0X0VUjgD+c6CzSLLkFZ02vA52WJe44iWyUTlJ6XqzVBe9eSdXvlXQ4blre89XzvTK7+25z8ztk1fU1KSWp44zy83Wpm3cVWrSs1Q1r825AY77lq4/zeUlHOpP2gw7kgCAAAAAGLhjiQAAACAmsNkO8lwRxIAAAAAEAt3JOux3R+nFq2vJOtMhjVa1NQrl0tm8vZ97oisT7LWYyHzlWFRazVKxV1nMttY0u3Xenr2RmnIOSYz5+t2RTtW580XR9Yv+HRjr5yemcyWl8yWiQzr1HRpzm3LOTOZbuJnXYs4kvJVyFxkHN/JUPYtzTjybfiv/Wxi+tfV7p1FRRtHOBMZtryumVd+6uS9Itsfe9eziceE/JiyJHoR23AGcuqEHhlaViHuSCbCHUkAAAAAQCwlu5A0s0vN7DMzmxz8GZJW9xszm2lm75nZIaUaIwAAAADgu0r9aOvVzrkr0zeYWW9JJ0jqI2kLSf81s+2cc9W/TgMAAACAomCynWRKfSFZnyMl3eucWylptpnNlDRQ0muFOmA+M5FJjz3rjg3Ppa/5tFWivsOZyY26Llv/etWyJon6DsuWgyyUJPlKSerV9IuMdee8f6JX/n7nmZF9xcktZjNmzg5e+Zre93rlQ9u+nbdjhbVptCJj3fNLe8fqa9f2H3vlW5/bzyv336Xy16YqZiYyrk5dv/LK+23xQYlGUjzNPo/+sTbhc//fsPX5ppDDqUi2ptQjyM2xt1zolR8848oMLYvrsJfOjazf7PGmkfXplvTt4JXXtLAMLVM2efNLrxy1bmQ2v+vxRGT9U6HyefeObvCxakV6VnGndp8VrO98q6nMJGIpdUbyXDObYma3m9m6WSG6SPo0rc2cYBsAAAAAJOck1bni/6kiBb2QNLP/mtk79fw5UtJNkraR1F/SPEn/WLdbPV3V+103szPMbIKZTVi7bFl9TQAAAAAAeVbQR1udcwfm0s7M/iXp8aA4R1L6nOtbSpqbof9bJN0iSc26dq2uS3wAAAAAhcPVQyIly0ia2ebOuXWLFh0t6Z3g9WOS7jazq5SabKenpEQL
```

Comment:
Is it possible that there are more complex transformations that could only be done by `groupby().apply()`? If so, you could end, "Nevertheless, you may find yourself in a situation where you have to apply a more complex transformation than this one that can't be done in the way we did above, and then `groupby().apply()` might be the way to go."

Reply:
Yes, I couldn't think of any right now. So I wasn't sure if this whole section was worth it on this blog. I think `groupby().apply()` can be the subject of several blogs all on its own! It seems this `Extension` section is weak and might be best taken out?